plugins/alerting/pkg/alerting/ops/ops.go

```diff
@@ -163,3 +166,22 @@ func (a *AlertingOpsNode) GetAvailableEndpoint(ctx context.Context, options *sha
 	}
 	return availableEndpoint, nil
 }
+
+func (a *AlertingOpsNode) GetAvailableCacheEndpoint(ctx context.Context, options *shared.AlertingClusterOptions) (string, error) {
+	var availableEndpoint string
+	status, err := a.GetClusterConfiguration(ctx, &emptypb.Empty{})
+	if err != nil {
+		return "", err
+	}
+	if status.NumReplicas == 1 { // exactly one that is the controller
+		availableEndpoint = options.GetControllerEndpoint()
+	} else {
+		availableEndpoint = options.GetWorkerEndpoint()
+	}
+	addr, _ := strings.CutPrefix(availableEndpoint, "http://")
```

Comment:
```suggestion
	_, addr, _ := strings.Cut(availableEndpoint, "://")
```

Reply:
when run in kubernetes this cuts the service name out of the endpoint, I'll keep it as `http://` since it is more prudent